### PR TITLE
Improved Memory Management.

### PR DESCRIPTION
**Pull request by simonmika**
Created new API for working with memory.

### CODE REVIEW

**Comment by simonmika** [comment]
@marcusnaslund Can you peer rieview this?

File: source/sdk/Buffer.ooc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	operator []= (start: Int, data: This) {
+		data copyTo(this[start])
+	}
+	operator []= (range: Range, data: This) {
+		data copyTo(this[range])
+	}
+	operator == (other: This) -> Bool {
+		(this _size == other _size) && (memcmp(this _pointer, other _pointer, this _size) == 0)
+	}
+	empty: static This { get { This new() } }
```
Comment (marcusnaslund) [inline source/sdk/Buffer.ooc]: We generally keep all properties at the top of the class.

**Comment by marcusnaslund** [comment]
Looks good. We might also want something to resize the `Buffer` though, as a replacement for `realloc`?

File: source/sdk/Buffer.ooc (hunk 1 above)
Comment (simonmika) [inline source/sdk/Buffer.ooc]: And all static stuff at the bottom.

**Comment by simonmika** [comment]
Feel free to add a `resize` method.

**Comment by simonmika** [comment]
@marcusnaslund Just because this can be better there are no need to stop this PR from merging. Don't let the perfect solution become the enemy of the good as long as the good leads to the perfect.

**Comment by marcusnaslund** [comment]
No, but I *am* stopping this PR because it is failing tests.

**Comment by simonmika** [comment]
Good.

**Comment by marcusnaslund** [comment]
I will take it from here in a new PR.

**Pull request closed without merging**